op-node/rollup/sync/start.go

```diff
@@ -133,7 +136,7 @@ func FindL2Heads(ctx context.Context, cfg *rollup.Config, l1 L1Chain, l2 L2Chain
 		// Fetch L1 information if we never had it, or if we do not have it for the current origin.
 		// Optimization: as soon as we have a previous L1 block, try to traverse L1 by hash instead of by number, to fill the cache.
 		if n.L1Origin.Hash == l1Block.ParentHash {
-			b, err := l1.L1BlockRefByHash(ctx, n.L1Origin.Hash)
+			b, err := fetchBlockWithHTTPRetry(func() (eth.L1BlockRef, error) { return l1.L1BlockRefByHash(ctx, n.L1Origin.Hash) }, lgr)
```

Comment:
We have a way to do this already, there is a client with retries built into it. We should make sure to use that instead of introducing something new. Its possible we may already be using it here

Reply:
If we are not doing this in the client, we should use the retry package which exists in `op-service` rather than re-implementing the retry logic.

Reply:
replaced with `op-service/retry`, please take another look